content/en/blog/general/automating-cis-compliance/index.md

```diff
@@ -0,0 +1,427 @@
+
+![Architecture Diagram](./cis-exec.png)
+
+- **Kyverno** (CNCF Incubating): Kubernetes-native policy engine for runtime validation
+- **kube-bench**: Community-standard CIS compliance scanner for node-level validation
+- **OpenTofu**: Open-source infrastructure-as-code for plan-time validation
+- **Kind**: CNCF-aligned local Kubernetes testing
+
+## Solution: Hybrid Cloud Native Policy Automation Architecture
+
+The framework implements a multi-layered approach combining the strengths of different CNCF and cloud native security tools:
+
+![Architecture Diagram](./cis-arch.png)
+
+### 1. Plan-Time Validation (Shift-Left Security):
+- Validates OpenTofu configurations before deployment
+- Catches misconfigurations early in development
+- Policies located in [policies/opentofu/](https://github.com/ATIC-Yugandhar/cis-eks-kyverno/blob/main/policies/opentofu)
+
+### 2. Runtime Validation (Continuous Monitoring):
+- **Kyverno**: CNCF Incubating project for Kubernetes resource validation
+- **Coverage**: Pod Security Standards, RBAC, Service Accounts, Network Policies
+
+### 3. Node-Level Validation (Deep System Inspection):
+- **kube-bench**: Industry-standard CIS compliance scanner with privileged access
+- **Coverage**: File permissions, kubelet configuration, control plane settings
+
+## Implementation: Complete Multi-Layer Security Pipeline
+
+### Step 1: Infrastructure Creation with OpenTofu
+
+Through building this open-source project, I've implemented a comprehensive security validation pipeline that demonstrates the entire infrastructure and application lifecycle - from infrastructure planning with OpenTofu to runtime enforcement with Kyverno and system-level auditing with kube-bench.
+
+*Note: This implementation has been thoroughly tested using KIND clusters and is available for community experimentation and contribution.*
+
+**File:** [opentofu/compliant/main.tf](https://github.com/ATIC-Yugandhar/cis-eks-kyverno/blob/main/opentofu/compliant/main.tf)
+
+```hcl
+resource "aws_eks_cluster" "main" {
+  name     = var.cluster_name
+  role_arn = aws_iam_role.eks.arn
+  version  = var.cluster_version
+
+  vpc_config {
+    subnet_ids              = aws_subnet.private[*].id
+    endpoint_private_access = true
+    endpoint_public_access  = false  # CIS 5.4.2 compliance
+  }
+
+  # CIS 2.1.1: Enable comprehensive audit logging
+  enabled_cluster_log_types = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
+
+  # CIS 5.3.1: Encrypt secrets with KMS
+  encryption_config {
+    resources = ["secrets"]
+    provider {
+      key_arn = aws_kms_key.eks.arn
+    }
+  }
+}
+```
+
+### Step 2: Plan-Time Policy Validation (Shift-Left Security)
+
+Before any infrastructure is deployed, OpenTofu configurations are validated against CIS policies using Kyverno-JSON for plan-time validation:
```

Comment:
Is there a reason why we used Kyverno-JSON and not the CEL based ValidatingPolicy?

Both will work, but one advantage of using ValidatingPolicy is that the same policy can be used for JSON payloads and Kubernetes resources. This makes it easier for users to maintain the policies.